qmt/materials/mat_data.py

```diff
@@ -0,0 +1,62 @@
+from ..geometry.geo_data_base import GeoData
+from qmt.materials import Materials
+from qmt.geometry import Geo3DData, Geo2DData
+from typing import Dict
+from qmt.infrastructure import WithParts
+from dataclasses import dataclass
+
+
+@dataclass
+class MatPart:
+    """Object containing material information for a part.
+    Currently only contains one string. Might contain more things in the future.
+
+    Parameters
+    ----------
+    material : str
+        The material for the part
+    """
+
+    material: str
+
+
+class MatData(WithParts):
+    def __init__(
+        self, materials_database: Materials, materials_mapping: Dict[str, str] = None
+    ):
+        """Class that contains materials information.
+
+        Parameters
+        ----------
+        materials_database : Materials
+            The materials database to use.
+        materials_mapping : dict
+            Mapping of parts to materials.
+            (Default = None)
+        """
+        self.materials_database = materials_database
+        if materials_mapping is not None:
+            super().__init__(
+                {name: MatPart(mat) for name, mat in materials_mapping.items()}
+            )
+
+    def get_material(self, part_name: str):
```

Comment:
I understand (and agree with) the reasoning behind not making material a part attribute. By why shouldn't we be able to call MatData.part_name to retrieve material data about a part? Using a getter seems a little clunky.

Reply:
Oh I just copied this from the old code. I can change it so you can do `mat_data[part_name]` now